### PR TITLE
Lesson 3 submission

### PR DESCRIPTION
**Pull request by emburyj**
Submitting Assignment 1 and updating lesson03 exercises

### CODE REVIEW

File: students/josh_embury/lesson03/mailroom.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
+lst_donor_table = [
+['Henry Michalson',10,500,25],
```
Comment (charlesrobison) [inline students/josh_embury/lesson03/mailroom.py:2]: Need spaces after commas, here and below in each list element

File: students/josh_embury/lesson03/mailroom.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    """
+    choice = str(input("Which option would you like to perform? [1 to 3] - ")).strip()
+    print()  # Add an extra line for looks
+    return choice
+
+def sendThankYou():
+    lst_donors = []
+    for current_donor in lst_donor_table:
+        lst_donors.append(current_donor[0])
+    donor =''
```
Comment (charlesrobison) [inline students/josh_embury/lesson03/mailroom.py:36]: Missing space after operator

File: students/josh_embury/lesson03/mailroom.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    while(len(donor)<1):
```
Comment (charlesrobison) [inline students/josh_embury/lesson03/mailroom.py:37]: Need spaces around operator

File: students/josh_embury/lesson03/mailroom.py [hunk 4]
```diff
@@ -0,0 +1,79 @@
+lst_donor_table = [
+['Henry Michalson',10,500,25],
+['Phil Hutch',76],
+['Galileo Humpkins',22000,100,490],
+['Methusela Honeysuckle',18,69,76000],
+['Lavender Goombs',55000,25],
+['test',1]
+]
+def showMenu():
```
Comment (charlesrobison) [inline students/josh_embury/lesson03/mailroom.py:9]: All function names in this code should be snake_case: show_menu

File: students/josh_embury/lesson03/mailroom.py [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+def formatRow(info_list):
+    return '{:<26} {:<2} {:10.2f} {:>11} {:^2} {:10.2f}'.format(info_list[0],info_list[1],info_list[2],info_list[3], info_list[4], info_list[5])
+def createReport():
+    header = '{:<25} {:^10} {:^10} {:^10}'.format('Donor Name', '| Total Given', '| Num Gifts', '| Average Gift')
+    print(header)
+    print('------------------------------------------------------------------')
+    for current_donor in lst_donor_table:
+        donor_data = getDonorData(current_donor[0])
+        new_row = [current_donor[0], '$', donor_data[0], donor_data[1], "$", donor_data[2]]
+        print(formatRow(new_row))
```
Comment (charlesrobison) [inline students/josh_embury/lesson03/mailroom.py:67]: Missing sorting function - report should be sorted by total given, in descending order

File: students/josh_embury/lesson04/dict_lab.py [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+#Dictionaries 2:
+dict2 = {}
+for dicval in dict1:
+    dict2[dicval] = dict1[dicval].lower().count('t')
+print('Creating new dict with t count:')
+for val in dict2.items():
+    print(val)
+print()
+#Sets
+s2=set()
```
Comment (charlesrobison) [inline students/josh_embury/lesson04/dict_lab.py:39]: Need spaces here and below too, around assignment operator

File: students/josh_embury/lesson04/dict_lab.py [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+s3=set()
+s4=set()
+for i in range(0,21):
+    if i%2 == 0:
```
Comment (charlesrobison) [inline students/josh_embury/lesson04/dict_lab.py:43]: Spaces needed around modulo operator, here and below

File: students/josh_embury/lesson04/mailroom2.py [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+    4: quitProgram
+}
+
+#-----------------------------------------------#
+# Main
+#-----------------------------------------------#
+if __name__ == '__main__':
+    while(True):
+        showMenu()  # Displays menu to console
+        dictMenu.get(int(getUserChoice()))()  # Get menu option by dict switch
```
Comment (charlesrobison) [inline students/josh_embury/lesson04/mailroom2.py:112]: Same feed back for prior version of mailroom, but this one is missing a couple of additional elements.  Lost the ability to 'list' existing donors and the exit command (4) isn't functioning correctly.

**Review by charlesrobison — APPROVED** [review]
A few elements missing from both versions of mailroom.  Take a look and let me know if you have questions.  Also, please submit all of these on the course site for grading.

Ready to merge

File: students/josh_embury/lesson03/mailroom.py (hunk 1 above)
Comment (emburyj) [reply students/josh_embury/lesson03/mailroom.py:2]: updated

File: students/josh_embury/lesson03/mailroom.py (hunk 2 above)
Comment (emburyj) [reply students/josh_embury/lesson03/mailroom.py:36]: updated

File: students/josh_embury/lesson03/mailroom.py (hunk 3 above)
Comment (emburyj) [reply students/josh_embury/lesson03/mailroom.py:37]: updated

File: students/josh_embury/lesson03/mailroom.py (hunk 4 above)
Comment (emburyj) [reply students/josh_embury/lesson03/mailroom.py:9]: updated

File: students/josh_embury/lesson03/mailroom.py (hunk 5 above)
Comment (emburyj) [reply students/josh_embury/lesson03/mailroom.py:67]: updated to sort donor table

File: students/josh_embury/lesson04/dict_lab.py (hunk 6 above)
Comment (emburyj) [reply students/josh_embury/lesson04/dict_lab.py:39]: updated

File: students/josh_embury/lesson04/dict_lab.py (hunk 7 above)
Comment (emburyj) [reply students/josh_embury/lesson04/dict_lab.py:43]: updated

File: students/josh_embury/lesson04/mailroom2.py (hunk 8 above)
Comment (emburyj) [reply students/josh_embury/lesson04/mailroom2.py:112]: Hi Charles, I've updated the mailroom2 script per the comments above. I'm not sure what you mean by "ability to 'list' existing donors." If you mean the Generate Report (2) function, this appears to be working. Also, the exit command seems to be working fine for me too. Can you show what errors you're having when running? Thank you, Josh

Comment (charlesrobison) [reply students/josh_embury/lesson04/mailroom2.py:112]: Hi Josh - when adding a donor or donation the user should be able to type 'list' to see a list of the existing donors in the db.  Your previous version had this feature, but it was omitted from this version.

**Pull request merged**